### PR TITLE
test(ticdc): migrate test-infra to testify for pkg/util pkg

### LINKED ISSUE #4879: migrate test-infra to testify for `pkg/util` pkg

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?

Issue Number: close #4879

### What is changed and how it works?
This pr migrate test-infra to testify for pkg/util pkg and and fix files that depend on `testleak.AfterTest` method calls

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update key monitor metrics in both TiCDC document and official document

### Release note 

```release-note
Please add a release note.
If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: cdc/sink/causality_test.go [hunk 1]
```diff
@@ -215,6 +213,6 @@ func (s *testCausalitySuite) TestGenKeys(c *check.C) {
 		sort.Slice(keys, func(i, j int) bool {
 			return bytes.Compare(keys[i], keys[j]) > 0
 		})
-		c.Assert(keys, check.DeepEquals, tc.expected)
+		require.Equal(t, keys, tc.expected)
```
Comment (hi-rustin) [inline cdc/sink/causality_test.go:216]:
```suggestion
		require.Equal(t, c.expected, keys)
```

The old and new assertion functions have different signatures, so please make sure you use the correct order for the rest as well.

**Review by hi-rustin — CHANGES_REQUESTED** [review]

**Comment by hi-rustin** [comment]
Thanks for working on this!

File: cdc/sink/causality_test.go (hunk 1 above)
Comment (mikechengwei) [reply cdc/sink/causality_test.go:216]: ok

**Comment by hi-rustin** [comment]
Also please refine your PR body. Do not submit the template directly as is. You will need to fill in and modify the contents of it.

**Comment by mikechengwei** [comment]
@hi-rustin hi, I found that the `mq_test.go` changes were relatively small, but some unknown errors occurred. Can you help me see?

**Comment by hi-rustin** [comment]
> @hi-rustin hi, I found that the `mq_test.go` changes were relatively small, but some unknown errors occurred. Can you help me see?

I think it's because these tests just haven't been run! mqSinkSuite does not appear to have hook up gocheck into the "go test" runner.

Also, the reason it fails looks like it's because we're updating the resloved ts asynchronously now. So we can't just get the value. You can try adding a delay to the test or adjusting the flush interval in the test.

**Comment by hi-rustin** [comment]
I would also suggest that you split your PRs, as oversized PRs can easily conflict and are not easily reviewed.
I would suggest starting with a few smaller packages.

**Comment by zhaoxinyu** [comment]
Avoid using testify/suite as much as possible. See https://github.com/pingcap/tiflow/issues/2164.
> Avoid using suite of testify as discussed errno: https://github.com/pingcap/tidb/pull/26082#issuecomment-877230972

**Comment by ben1009** [comment]
could all the change ` Files changed 34` separate into small prs instead ?

**Comment by overvenus** [comment]
Really appreciate your work!
Could you split this PR into smaller PRs? It helps code review and speed up merging (by less confliction and less rebase).

**Comment by mikechengwei** [comment]
ok

**Comment by mikechengwei** [comment]
> I think it's because these tests just haven't been run! mqSinkSuite does not appear to have hook up gocheck into the "go test" runner.
> 
> Also, the reason it fails looks like it's because we're updating the resloved ts asynchronously now. So we can't just get the value. You can try adding a delay to the test or adjusting the flush interval in the test.

No problem.I will split this pr.

**Comment by mikechengwei** [comment]
Close this pr,create new pr #5036 .

**Pull request closed without merging**